### ISSUE TITLE
EDX Quantification for SEM

### ISSUE THREAD

**Issue by k8macarthur** [hyperspy/hyperspy#2332]
I've noticed some interest in SEM EDX quantification. I think this would be a nice addition to Hyperspy. Currently the k-factor quantification for TEM would be directly transferable. The critical step would be to add a ZAF correction. 

Unfortunately, not being an SEM expert I'm not exactly sure what the ZAF correction entails.

**Comment by gegelagaffe** [comment]
To be honest, I'll have to do it for my own research... I'm chemo-physics researcher, I'm not an expert of SEM but I use it more or less every day. I clearly have a lack of knowledge in hysperspy compare to you so If you have the whish and the time to help me on this topic, it would be very helpful.

**Comment by k8macarthur** [comment]
Hi! That's great to hear. I assume some direct copy of the TEM quantification with k-factors and absorption correction would be a good place to start. I'm assuming atomic number and fluorescence are also iterative corrections. Do you know this? I'm happy to set up the main outline of the function if you could find the specific maths for SEM? Then we would need to add tests and documentation after.

**Comment by k8macarthur** [comment]
https://github.com/hyperspy/hyperspy/blob/851c9c0687533f429c853873834100aae2e6f92b/hyperspy/misc/eds/utils.py

See here for the location of the k-factor quantification for EDX TEM

**Comment by gegelagaffe** [comment]
I agree the TEM quantification process is a good start. Concerning the specific Maths equation for SEM I found it :
https://laboutique.edpsciences.fr/produit/9782759800827
This book is a bible of SEM quantification, I'll need a week to read the quantification chapter and understand it. After that I will send you a summurize of the ZAF correction process.

I'm working on my SEM to obtain standards spectra essential for the SEM quantification. 

When the script will work, I have several cases to test the quantification.

So first give me a week to improve my skill in  ZAF correction equation ;)


Thanks again to help me on this project

**Comment by k8macarthur** [comment]
@gegelagaffe I know a lot has been happening in the last few months but I was just wondering if  you had a chance to look at this yet?

**Comment by gegelagaffe** [comment]
I made a lot... It's almost done ( even if there is still a lot to do)... To be honest I found another way than quantification to solve my problem with the phases recognition... May be I'll could show you what I'm now able to do with teams or jitsy meeting ?

Envoyé d’Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by k8macarthur** [comment]
@gegelagaffe I sent you a message on gitter but maybe it didn't get through. I'm happy to chat and see what you've managed. Maybe respond to my gitter message with what times would suit. I'm free Wednesday and Friday this week.
